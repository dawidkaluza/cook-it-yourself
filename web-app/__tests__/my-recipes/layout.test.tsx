import {render, screen} from "@testing-library/react";
import Layout from "@/app/my-recipes/layout";

vi.mock("next/navigation", () => {
  return {
    usePathname: () => "/my-recipes/",
  };
});

describe("layout component", () => {
  test("renders", () => {
    // Given, when
    const layout = render(<Layout><p>Hello World</p></Layout>);

    // Then
    const navigation = screen.getByRole("navigation");
    expect(navigation).not.toBeNull();

    const text = screen.getByText(/hello world/i);
    expect(text).not.toBeNull();

    expect(layout.container).toMatchSnapshot();
  });
});