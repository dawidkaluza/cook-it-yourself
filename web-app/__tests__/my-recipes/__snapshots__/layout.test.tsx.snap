// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`layout component > renders 1`] = `
<div>
  <nav
    class="m-3"
  >
    <ol
      class="breadcrumb"
    >
      <li
        class="breadcrumb-item"
      >
        <a
          href="/my-recipes"
        >
          My recipes
        </a>
      </li>
      <li
        class="breadcrumb-item active"
      />
    </ol>
  </nav>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <p>
          Hello World
        </p>
      </div>
    </div>
  </div>
</div>
`;
